basilisk/src/ast/references.c

```
@@ -0,0 +1,333 @@
+
+  if (!attributes && !ast_parent (ref, sym_compound_statement) && !ast_parent (ref, sym_parameter_declaration))
+    str_append (references, ",.global=1");
+  
+  /**
+  Assumes 'double *' are references to arrays with 'nl'
+  elements. Fixme: this is very specific and should be made more
+  general e.g. systematically using 'fat pointers' to get array
+  sizes. */
+
+  Ast * nl;
+  if (type->sym == sym_DOUBLE && dim.pointer == 1 && !dim.dimension && strcmp (ast_terminal (ref)->start, "_constant") &&
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)